### ISSUE TITLE
Why this module uses "data" and not "resource" for IAM policy document?

### ISSUE THREAD

**Issue by Defozo** [cloudposse/terraform-aws-ecs-codepipeline#11]
I'm a newbie in Terraform world and I'd like to ask, why this module assumes a user has appropriate IAM policy document? IMHO it should create one.

Relevant code:

```
data "aws_iam_policy_document" "assume" {
  statement {
    sid = ""

    actions = [
      "sts:AssumeRole",
    ]

    principals {
      type        = "Service"
      identifiers = ["codepipeline.amazonaws.com"]
    }

    effect = "Allow"
  }
}
```

**Comment by aknysh** [comment]
@Defozo 
good question!

The module creates the IAM role here https://github.com/cloudposse/terraform-aws-ecs-codepipeline/blob/master/main.tf#L28

`data "aws_iam_policy_document" "assume"`  does not say a user has a policy document, it's just a Terraform way to create policy documents instead of using JSON files.
This is used here https://github.com/cloudposse/terraform-aws-ecs-codepipeline/blob/master/main.tf#L31, converting it to JSON and specifying a policy to allow `codepipeline.amazonaws.com` service to assume the created IAM role.

Same is here https://github.com/cloudposse/terraform-aws-ecs-codepipeline/blob/master/main.tf#L63 which creates a policy document, then creates a policy resource here https://github.com/cloudposse/terraform-aws-ecs-codepipeline/blob/master/main.tf#L60, then attaches the policy to the role here https://github.com/cloudposse/terraform-aws-ecs-codepipeline/blob/master/main.tf#L86

In short, you create:
1. A policy document using `data "aws_iam_policy_document"` (which is just another format for JSON files)
2. A policy from the policy document  
3. Attach the policy to the role

Thanks
(will close for now, please reopen if you have more questions)
